app/src/main/kotlin/co/touchlab/droidconandroid/UserDetailFragment.kt

```diff
@@ -38,33 +42,32 @@ private const val LINKEDIN_PREFIX: String = "http://www.linkedin.com/in/"
 private const val FACEBOOK_PREFIX: String = "http://www.facebook.com/"
 private const val PHONE_PREFIX: String = "tel:"
 
-class UserDetailFragment() : Fragment()
-{
-    companion object
-    {
+class UserDetailFragment() : Fragment(), UserDetailHost {
+
+    private lateinit var presenter: UserDetailPresenter
+
+    companion object {
         val TAG: String = UserDetailFragment::class.java.simpleName
 
-        interface FinishListener
-        {
+        interface FinishListener {
             fun onFragmentFinished()
         }
     }
 
-    override fun onCreate(savedInstanceState: Bundle?)
-    {
+    override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        EventBusExt.getDefault().register(this)
-        TaskQueue.loadQueueNetwork(activity).execute(FindUserTask(findUserCodeArg()))
+        val helper = DatabaseHelper.getInstance(activity)
+        val restAdapter = DataHelper.makeRequestAdapter(activity, AppManager.getPlatformClient())
+        val task = FindUserTask(helper, restAdapter)
+        presenter = UserDetailPresenter(this, task)
+        presenter.findUser(findUserCodeArg())
     }
 
-
-    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View?
-    {
-        return inflater !!.inflate(R.layout.fragment_user_detail, null)
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```

Comment:
OPTIONAL: Its a minor thing, but I mentioned it in Sufei's change too. When android studio adds these functions it includes the '?' on all of the parameters, but they dont all strictly require it. So you can remove the '?' on the inflater here and on the view in onViewCreated so that you dont have to do the '!!' when you use them